### PR TITLE
media: Remove duplicated code

### PR DESCRIPTION
**Pull request by junmin-kim**
remove duplicated code using inheritance

### CODE REVIEW

**Comment by junmin-kim** [comment]
@jsdosa I need your opinion

**Comment by seinfra** [comment]
Target : [725e10d3a4f8dcb0a485ef2e2bfc5ce573534645] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [725e10d3a4f8dcb0a485ef2e2bfc5ce573534645] - Code Rule Check Result:
framework/src/media/MediaWorker.h:34: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/MediaWorker.h:46: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaWorker.h:46: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/MediaWorker.h:46: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/MediaWorker.h:47: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/MediaWorker.h:49: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/PlayerObserverWorker.h:35: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/src/media/PlayerObserverWorker.h:38: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/PlayerWorker.h:29: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/src/media/PlayerWorker.h:36: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/PlayerWorker.h:36: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/PlayerWorker.h:36: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/PlayerWorker.h:36: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/src/media/PlayerWorker.h:37: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/PlayerWorker.h:37: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/PlayerWorker.h:37: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/PlayerWorker.h:37: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/src/media/PlayerWorker.h:38: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/PlayerWorker.h:38: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/PlayerWorker.h:38: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/PlayerWorker.h:38: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/src/media/PlayerWorker.h:46: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/PlayerWorker.h:46: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/PlayerWorker.h:46: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)
framework/src/media/RecorderObserverWorker.h:38: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/src/media/RecorderObserverWorker.h:43: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/RecorderWorker.h:40: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
framework/src/media/RecorderWorker.h:43: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/RecorderWorker.h:43: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/RecorderWorker.h:43: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/RecorderWorker.h:43: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/src/media/RecorderWorker.h:44: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/RecorderWorker.h:44: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/RecorderWorker.h:44: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/RecorderWorker.h:44: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxO)
framework/src/media/RecorderWorker.h:45: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/src/media/RecorderWorker.h:45: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/RecorderWorker.h:45: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/RecorderWorker.h:45: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxB)
framework/src/media/RecorderWorker.h:49: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/src/media/RecorderWorker.h:58: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/src/media/RecorderWorker.h:58: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:VxV)
framework/src/media/RecorderWorker.h:58: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:VxW)

**Comment by junmin-kim** [comment]
Please DO NOT MERGE

File: framework/src/media/Make.defs [hunk 1]
```diff
@@ -27,7 +27,7 @@ endif
 CFLAGS += -D__TINYARA__
 
 ifeq ($(CONFIG_MEDIA), y)
-CXXSRCS += MediaQueue.cpp DataSource.cpp MediaUtils.cpp
+CXXSRCS += MediaQueue.cpp DataSource.cpp MediaUtils.cpp MediaWorker.cpp
```
Comment (Taejun-Kwon) [inline framework/src/media/Make.defs]: good.

File: framework/src/media/RecorderObserverWorker.cpp [hunk 2]
```diff
@@ -19,9 +19,8 @@
 #include "RecorderObserverWorker.h"
 
 namespace media {
-once_flag RecorderObserverWorker::mOnceFlag;
 
-RecorderObserverWorker::RecorderObserverWorker() : mRefCnt(0)
+RecorderObserverWorker::RecorderObserverWorker()
```
Comment (Taejun-Kwon) [inline framework/src/media/RecorderObserverWorker.cpp]: Player using std::lock_guard, but recorder using std::unique_lock.
Why they are diffrent?

File: framework/src/media/PlayerObserverWorker.cpp [hunk 3]
```diff
@@ -64,18 +66,15 @@ void PlayerObserverWorker::stopWorker()
 {
 	std::lock_guard<std::mutex> lock(mRefMtx);
 	medvdbg("PlayerObserverWorker : stopWorker\n");
-	if (--mRefCnt <= 0) {
+	decreaseRef();
+
+	if (mRefCnt <= 0) {
```
Comment (Taejun-Kwon) [inline framework/src/media/PlayerObserverWorker.cpp]: we don't have to check joinable here?

**Pull request closed without merging**